ci/parse.nix

```diff
@@ -0,0 +1,45 @@
+{
+  fd,
+  lib,
+  nix,
+  runCommand,
+}:
+let
+  nixpkgs =
+    with lib.fileset;
+    toSource {
+      root = ../.;
+      fileset = (fileFilter (file: file.hasExt "nix") ../.);
+    };
+in
+runCommand "nix-parse-${nix.name}"
+  {
+    nativeBuildInputs = [
+      fd
+      nix
+    ];
+  }
+  ''
+    export NIX_STORE_DIR=$TMPDIR/store
+    export NIX_STATE_DIR=$TMPDIR/state
+
+    cd "${nixpkgs}"
+
+    # Passes all files to nix-instantiate at once with -X.
+    # Much faster, but will only show first error.
+    parse-all() {
+      fd -t file -e nix -X nix-instantiate --parse >/dev/null 2>/dev/null
```

Comment:
Does fd run programs in parallel? Otherwise xargs -P should be preferred.

Reply:
Seems slower for me than find + xargs for the -X case:

```
% time find . -type f -name '*.nix' | xargs -P $(nproc) nix-instantiate --parse >/dev/null 2>/dev/null
find . -type f -name '*.nix'  0.13s user 0.37s system 99% cpu 0.495 total
xargs -P $(nproc) nix-instantiate --parse > /dev/null 2> /dev/null  9.64s user 1.42s system 637% cpu 1.736 total
```

```
% time sh -c 'fd -t file -e nix -X nix-instantiate --parse >/dev/null 2>/dev/null'
sh -c 'fd -t file -e nix -X nix-instantiate --parse >/dev/null 2>/dev/null'  8.27s user 2.29s system 135% cpu 7.778 total
```

Reply:
The for-each version just needs an -n1:

```
% time find . -type f -name '*.nix' | xargs -n1 -P $(nproc) nix-instantiate --parse >/dev/null
error: syntax error, unexpected '=', expecting end of file
       at /home/joerg/git/nixpkgs/flake.nix:4:15:
            3|
            4|   description = "A collection of packages for the Nix package manager";
             |               ^
            5|
uj
```

Reply:
But it also looks we don't need to run it twice with xargs:

```
% time find . -type f -name '*.nix' | xargs -P $(nproc) nix-instantiate --parse >/dev/null
error: syntax error, unexpected '=', expecting end of file
       at /home/joerg/git/nixpkgs/flake.nix:4:15:
            3|
            4|   description = "A collection of packages for the Nix package manager";
             |               ^
            5|
find . -type f -name '*.nix'  0.12s user 0.37s system 99% cpu 0.494 total
```

That error looks fine to me.

Reply:
Yeah. That and my gut feeling tells me that fd will result in a bigger dependency closure than findutils.